src/App.tsx

```diff
@@ -26,70 +26,131 @@ type ReorderOptions = {
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByName = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const {
+      sortByName,
+      sortByLength,
+      reverse,
+      reset,
+    } = this;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              sortType !== SortType.ALPABET
+                ? 'button is-info is-light'
+                : 'button is-info'
+            }
+            onClick={sortByName}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              sortType !== SortType.LENGTH
+                ? 'button is-success is-light'
+                : 'button is-success'
+            }
+            onClick={sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={
+              isReversed !== true
+                ? 'button is-warning is-light'
+                : 'button is-warning'
+            }
+            onClick={reverse}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed === true)
+            && (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={reset}
+              >
+                Reset
+              </button>
+            )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {getReorderedGoods(goodsFromServer, this.state)
+              .map(good => <li data-cy="Good">{good}</li>)}
```

Comment:
```suggestion
            {getReorderedGoods(goodsFromServer, this.state)
              .map(good => <li data-cy="Good">{good}</li>)}
```
add key to li element

Reply:
> don't forget to add key, you already know why it's important 😉

yes, but I'm confused because the checklist for this task says not to generate a key on render. (So ​​I removed it :) )